tests/spin_targets.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+#
+#    Copyright 2024 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Module for testing different targets.
+Forked from tests/test_target.py.
+"""
+
+import os
+from unittest import TestCase
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget, QEMUTargetRunner
+from devlib.utils.misc import get_random_string
+from devlib.exception import TargetStableError
+
+
+class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self):
+        """ Test Target.read_tree_values_flat() """
+
+        def run_test(target):
+            data = {
+                'test1': '1',
+                'test2': '2\n\n',
+                'test3': '3\n\n4\n\n',
+            }
+
+            print(f'{method_name}: target={target.__class__.__name__} os={target.os} ' \
```

Comment:
generally it's better to avoid breaking a string literal, as this prevents from using grep easily when looking for the source of a given message